api/transport/stream_test.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package transport_test
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/api/transport/transporttest"
+)
+
+func TestServerStreamHeaders(t *testing.T) {
+	items := map[string]string{"header-key": "header-value"}
+	headers := transport.HeadersFromMap(items)
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockStream := transporttest.NewMockStream(mockCtrl)
+
+	t.Run("unimplemented", func(t *testing.T) {
+		serverStream, err := transport.NewServerStream(mockStream)
+		require.NoError(t, err)
```

Comment:
can we add messages to the require/asserts, in case anything fails?

Reply:
Is that the style? Unless I'm going to extract something to provide with custom formatting, a `"msg"` doesn't provide any more information than the `file:lineno` and `err` contents.

Reply:
yeah fair point 🆗 , disregard